.github/workflows/main.yml

```diff
@@ -0,0 +1,209 @@
+          skip_ssh_tests: true
+          skip_negotiate_tests: true
+        - # Windows amd64 Visual Studio
+          name: Windows
+          arch: amd64
+          cmake_generator: Visual Studio 16 2019
+          os: windows-latest
+          cmake_options: -A x64 -DMSVC_CRTDBG=ON -DDEPRECATE_HARD=ON
+          skip_ssh_tests: true
+          skip_negotiate_tests: true
+        - # Windows x86 Visual Studio
+          name: Windows 2019
+          arch: x86
+          cmake_generator: Visual Studio 16 2019
+          os: windows-latest
+          cmake_options: -A Win32 -DMSVC_CRTDBG=ON -DDEPRECATE_HARD=ON -DUSE_SHA1=HTTPS
+          skip_ssh_tests: true
+          skip_negotiate_tests: true
+        - # Windows amd64 mingw
+          name: Windows
+          arch: amd64
+          cmake_generator: MinGW Makefiles
+          os: windows-latest
+          setup: setup-mingw.sh
+          build_temp: D:\Temp
+          build_path: D:\Temp\mingw64\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Program Files (x86)\CMake\bin
+          cmake_options: -DDEPRECATE_HARD=ON
+          skip_ssh_tests: true
+          skip_negotiate_tests: true
+        - # Windows x86 mingw
+          name: Windows
+          arch: x86
+          cmake_generator: MinGW Makefiles
+          os: windows-latest
+          setup: setup-mingw.sh
+          build_temp: D:\Temp
+          build_path: D:\Temp\mingw32\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Program Files (x86)\CMake\bin
+          cmake_options: -DDEPRECATE_HARD=ON
+          skip_ssh_tests: true
+          skip_negotiate_tests: true
+      fail-fast: false
+    # Turn matrix variables into environment variables.
+    env:
+      CC: ${{ matrix.platform.cc }}
+      ARCH: ${{ matrix.platform.arch }}
+      BUILD_TEMP: ${{ matrix.platform.build_temp }}
+      BUILD_PATH: ${{ matrix.platform.build_path }}
+      CMAKE_GENERATOR: ${{ matrix.platform.cmake_generator }}
+      CMAKE_OPTIONS: ${{ matrix.platform.cmake_options }}
+      PKG_CONFIG_PATH: ${{ matrix.platform.pkg_config_path }}
+      SKIP_SSH_TESTS: ${{ matrix.platform.skip_ssh_tests }}
+      SKIP_NEGOTIATE_TESTS: ${{ matrix.platform.skip_negotiate_tests }}
+      GITTEST_NEGOTIATE_PASSWORD: ${{ secrets.GITTEST_NEGOTIATE_PASSWORD }}
+    runs-on: ${{ matrix.platform.os }}
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+    - name: Set up build environment
+      run: azure-pipelines/${{ matrix.platform.setup }}
+      shell: bash
+      if: matrix.platform.setup != ''
+    - name: Build and test
+      run: |
+        if [ -n "${{ matrix.platform.image }}" ]; then
+          docker_container="${{ github.repository }}/${{ matrix.platform.image }}:latest"
+          docker login https://${{ env.docker-registry }} -u ${{ github.actor }} -p ${{ github.token }}
+          docker pull ${{ env.docker-registry }}/${docker_container}
```

Comment:
Is this explicit pull required? I thought `docker run` can be directly passed the image we want to run.